### PR TITLE
Depth post-processing filters, align depth and color frames, integrate laser detection ML model

### PR DESCRIPTION
**Pull request by genkikondo**
- Add depth frame post-processing filters
- Align depth frame to color frame
- Add SetExposure service to camera control
- Use laser detection ML model
- Set manual exposure during calibration
- Add COLCON_IGNORE to ML projects
- Fix parameters yaml

### CODE REVIEW

File: camera_control/camera_control/camera/realsense.py [hunk 1]
```diff
@@ -1,98 +1,132 @@
...
             return None
-        depth = frame["depth"].get_distance(
+        depth = frames["depth"].get_distance(
             round(depth_point[0]), round(depth_point[1])
         )
         pos_wrt_color = rs.rs2_deproject_pixel_to_point(
-            self.color_intrins, [x, y], depth
+            self.color_intrinsics, [x, y], depth
         )
         pos_wrt_color = np.array(pos_wrt_color) * 1000
```
Comment (genkikondo) [inline camera_control/camera_control/camera/realsense.py:107]: Why is there a *1000 here?

Comment (genkikondo) [reply camera_control/camera_control/camera/realsense.py:107]: I don't think it functionally affects anything as of now, but IMO we should stick to meters globally

**Comment by genkikondo** [comment]
Removed duplicate parameters file in top level config/ dir. Fixed parameters yaml (ros__parameters was missing an underscore). Fixed control_node launch file to properly use the parameters yaml, as the reference wasn't correct

**Comment by genkikondo** [comment]
Switched over to using laser detection ML model instead of heuristics, as I've found it  more robust to environmental conditions

File: camera_control/camera_control/camera/realsense.py (hunk 1 above)
Comment (CA-rvinson) [reply camera_control/camera_control/camera/realsense.py]: Thats fine by me, I'm a bit more partial to mm everywhere, but I don't really care.

**Comment by CA-rvinson** [comment]
> Switched over to using laser detection ML model instead of heuristics, as I've found it more robust to environmental conditions

Sounds great, I'm modifying the camera control node to allow us to choose different models we want to use, I think I'm gonna hold off on that until this merges to prevent any rebase/merge issues. In the mean time I'm gonna look at implementing torch metrics in particular https://torchmetrics.readthedocs.io/en/stable/detection/mean_average_precision.html

**Pull request merged**